Sources/xcproj/XCScheme.swift

```diff
@@ -169,9 +169,75 @@ final public class XCScheme {
         }
     }
 
+    final public class ExecutionAction {
+        public var title: String
+        public var scriptText: String
+        public var environmentBuildable: BuildableReference?
+
+        public init(scriptText: String, title: String? = nil, environmentBuildable: BuildableReference? = nil) {
+            self.scriptText = scriptText
+            self.title = title ?? "Run Script"
+            self.environmentBuildable = environmentBuildable
+        }
+
+        init(element: AEXMLElement) throws {
+            self.scriptText = element["ActionContent"].attributes["scriptText"] ?? ""
+            self.title = element["ActionContent"].attributes["title"] ?? "Run Script"
+            self.environmentBuildable = try? BuildableReference(element: element["ActionContent"]["EnvironmentBuildable"]["BuildableReference"])
+        }
+
+        private static let ActionType = "Xcode.IDEStandardExecutionActionsCore.ExecutionActionType.ShellScriptAction"
+
+        fileprivate func xmlElement() -> AEXMLElement {
+            let element = AEXMLElement(name: "ExecutionAction",
+                                       value: nil,
+                                       attributes: [ "ActionType": ExecutionAction.ActionType ])
+            let content = AEXMLElement(name: "ActionContent",
+                                       value: nil,
+                                       attributes: [ "title": title,
+                                                     "scriptText": scriptText ])
+            element.addChild(content)
+            if let environmentBuildable = environmentBuildable {
+                let environment = content.addChild(name: "EnvironmentBuildable")
+                environment.addChild(environmentBuildable.xmlElement())
+            }
+            return element
+        }
+    }
+
     // MARK: - Build Action
 
-    final public class BuildAction {
+    public class SerialAction {
```

Comment:
Are there any common properties that could go in here in the future? If so we could just call this `SchemeAction`

Reply:
I picked the name for what it is now, since I don't know if it will expand to be more. I figured it could easily be renamed. Having said all that, I can go either way, it's not something I feel strongly about.

Reply:
Yeah, it's fine now. Just thought I'd mention it